java/java.lsp.server/vscode/src/nbcode.ts

```diff
@@ -86,7 +88,8 @@ if (typeof process === 'object' && process.argv0 === 'node') {
     }
     let clusters = fs.readdirSync(nbcode).filter(c => c !== 'bin' && c !== 'etc').map(c => path.join(nbcode, c));
     let args = process.argv.slice(2);
-    let globalStorage = path.join(os.homedir(), '.config', 'Code', 'User', 'globalStorage', 'jlahoda.apache-netbeans-java');
+    let json = JSON.parse("" + fs.readFileSync(path.join(extension, 'package.json')));
+    let globalStorage = path.join(os.homedir(), '.config', 'Code', 'User', 'globalStorage', json.publisher + '.' + json.name);
```

Comment:
While at it, let also read the `publisher` and extension `name` from the actual `package.json` file.